docs/releases.rst

```diff
@@ -13,6 +13,10 @@ limit for each test.
 The provision plugins now support setting an ``environment`` key.
 See the documentation of :ref:`/plugins/provision`.
 
+The finish :ref:`/plugins/finish/shell` plugin now supports ``url`` and
+``ref`` options, so finish scripts can be fetched directly from
+remote git repositories.
```

Comment:
This message should probably begin by mentioning that `prepare/shell` and `finish/shell` now share the implementation and therefore the functionality as well. For example, the recently added support for `url` and `ref` keys (keys, not options, please) is now supported by `finish/shell` too.

Reply:
And, please, change the subject of this PR - it's not adding `ref` and `url `options to `shell` finish plugin, it's destroying that plugin and replaces its internals with code from `prepare`. In other words, it reuses `prepare/shell` implementation to power `finish/shell`, just like we reuse `prepare/ansible` in `finish/ansible`. `finish/shell` gaining new functionality is pretty much a happy side-effect, not the goal of this PR - the goal is the reduction of duplicity, and PR subject (and commit messages) should reflect that.

Reply:
Release note and PR summary + description updated!